### ISSUE TITLE
endpointslice labels not correct with UDN enabled and create failure retrys will cause alerts to fire

### ISSUE THREAD

**Issue by jluhrsen** [ovn-kubernetes/ovn-kubernetes#4896]
### What happened?

this is seen in openshift CI

essentially `OVNKubernetesResourceRetryFailure` alerts fire because some endpointslice creation fails and it hits enough retries to cause the alert to fire.

This is happening in the middle of the e2e tests and I'm guessing it will all be shown to be triggered by the same test, although I did not confirm that yet.

the failure in the log is this:

`Failed to create *factory.endpointSliceForGateway e2e-endpointslice-8038/e2e-example-inghvcwj, error: endpointslice e2e-endpointslice-8038/e2e-example-inghvcwj: empty value for label [kubernetes.io/service-name](http://kubernetes.io/service-name) in network default`

That comes from here.

I don't know how the eps slice creation is made or what labels should or should not be there when UDN is enabled, but
maybe when we are in UDN mode we neglect to add that kubernetes.io/service-name label? or maybe we don't want
that label and we need to update that ServiceFromEndpointSlice() to account for that.

[here](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2357/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-techpreview/1865136945068249088) is an example job.

### What did you expect to happen?

endpointslice creation should work w/ NetworkSegmentation enabled

### How can we reproduce it (as minimally and precisely as possible)?

use openshift CI

### OVN-Kubernetes version

24.09.0-33.el9fdp

### Kubernetes version

v1.31.1

### OVN version

<details>

```console
$ oc rsh -n ovn-kubernetes ovnkube-node-xxxxx (pick any ovnkube-node pod on your cluster)
$ rpm -q ovn
# paste output here
```

</details>

### OVS version

<details>

```console
$ oc rsh -n ovn-kubernetes ovs-node-xxxxx (pick any ovs pod on your cluster)
$ rpm -q openvswitch
# paste output here
```

</details>

### Platform

all

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

**Comment by jluhrsen** [comment]
/assign @kyrtapz 

@kyrtapz , I dug a little bit more and it looks like the test case causing this might be [this one](https://github.com/openshift/origin/blob/aaa0c2df6fce338d15ba884e449ff97766d0a92b/vendor/k8s.io/kubernetes/test/e2e/network/endpointslice.go#L365)

when that endpointslice hits the [util.go.ServiceFromEndpointSlice()](https://github.com/ovn-kubernetes/ovn-kubernetes/blob/2e184829ce00f9d86ad275fc40ce1d5ec7a57671/go-controller/pkg/util/util.go#L598-L614) I can see it has a single label key of "special-label" (saw that from [this debug commit](https://github.com/openshift/ovn-kubernetes/pull/2394/commits/0fa065adf312b3186507f395147098f1a0d3a399). [example job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2394/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-techpreview/1867277512447365120))

I don't see that label in a [non UDN job](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_ovn-kubernetes/2394/pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn/1867277483074654208) though. it looks like those e2e endpointslices end up with the kubernetes.io/managed-by and kubernetes.io/service-name labels.

**Comment by jluhrsen** [comment]
> @jluhrsen : Is something not clear hereon the fix #4902 ? @kyrtapz already has a fix to fix the EPS issue.... this issue is basically fixed tat's to his PR 4902, no need to dig more

yeah, what wasn't clear was that work was being done. I did not know a PR was being worked on and no comments happened in this issue so I spent more time trying to debug and figure out. :(

**Comment by jluhrsen** [comment]
> > @jluhrsen : Is something not clear hereon the fix #4902 ? @kyrtapz already has a fix to fix the EPS issue.... this issue is basically fixed tat's to his PR 4902, no need to dig more
> 
> yeah, what wasn't clear was that work was being done. I did not know a PR was being worked on and no comments happened in this issue so I spent more time trying to debug and figure out. :(

I totally missed the auto-message about the PR linked to this. whoops!
